src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -180,11 +180,18 @@ internal void AppendMetadata()
             }
         }
 
-        private void WriteMethods(TypeBuilderImpl typeBuilder, List<GenericTypeParameterBuilderImpl> genericParams)
+        private void WriteMethods(TypeBuilderImpl typeBuilder, List<GenericTypeParameterBuilderImpl> genericParams, MethodBodyStreamEncoder methodBodyEncoder)
         {
             foreach (MethodBuilderImpl method in typeBuilder._methodDefinitions)
             {
-                MethodDefinitionHandle methodHandle = AddMethodDefinition(method, method.GetMethodSignatureBlob(), _nextParameterRowId);
+                int offset = -1;
+                ILGeneratorImpl? il = method.ILGeneratorImpl;
```

Comment:
Check for `!method.IsAbstract` before setting variable for `method.ILGeneratorImpl;`?

Reply:
Do we even need `method.IsAbstract` check here? It does not look like right to throw away the IL body.

Reply:
Good point, the runtime implementation throws while populating the method bodies on `TypeBuilder.CreateTypeInfo()`:
https://github.com/dotnet/runtime/blob/087ca4662309f3db40b3096cbf89853d9c292e16/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs#L1696
As we are not populating the `TypeInfo` similarly, probably we should check it and throw in `MethodBuilder.GetILGenerator()`?

Reply:
I agree.